packages/clerk-js/src/utils/authPropHelpers.ts

```diff
@@ -112,3 +112,13 @@ export const buildAuthQueryString = (data: BuildAuthQueryStringArgs): string | n
   }
   return Object.keys(query).length === 0 ? null : qs.stringify(query);
 };
+
+export const normalizeRoutingOptions = ({
+  routing,
+  path,
+}: {
+  routing?: RoutingStrategy;
+  path?: string;
+}): RoutingOptions => {
+  return { routing: !!path && !routing ? 'path' : routing, path } as RoutingOptions;
```

Comment:
The `as` assertion here is a code-smell, it shouldn't be necessary. What happens when you remove it?

Reply:
It's necessary as the the type is a union and in the case `virtual` or `hash` the objects does not have the `path` property.

Reply:
Is the TS compiler happy if you don't use ternaries but a bit more verbose syntax?

```ts
if (!!path && !routing) {
return { routing: 'path', path }
} else {
  return { routing }
}
```

Reply:
Yeah I've tried that but still it's not happy